cborattr/src/cborattr.c

```diff
@@ -67,6 +67,11 @@ valid_attr_type(CborType ct, CborAttrType at)
         }
 	break;
 #if FLOAT_SUPPORT
+    case CborAttrHalfFloatType:
+        if (ct == CborHalfFloatType) {
+            return 1;
+        }
+        break;
```

Comment:
Since you are representing a half-float using an integer type, are you sure you need to add it under `FLOAT_SUPPORT`?

Reply:
Definitely yes. The half float type will only be used when CBOR is compacting float data into a half float type to save on message space.